### PR TITLE
Move /seccomp/ into domain prefix in seccomp annotations

### LINKED ISSUE #26610: Seccomp annotations name format need redefine
Now pod annotations validation only allow no more than one slash. But seccomp annotations format has more than one slash. like: security.alpha.kubernetes.io/seccomp/container/<container name>
https://github.com/kubernetes/kubernetes/blob/master/docs/design/seccomp.md#api-changes

When create a pod with  seccomp annotations, it will error.
```
$ kubectl create -f pod-seccomp.yaml
The Pod "pod-seccomp" is invalid.
metadata.annotations: Invalid value: "security.alpha.kubernetes.io/seccomp/pod": must be a qualified name (at most 63 characters, matching regex ([A-Za-z0-9][-A-Za-z0-9_.])?[A-Za-z0-9]), with an optional DNS subdomain prefix (at most 253 characters, matching regex [a-z0-9]([-a-z0-9][a-z0-9])?(.a-z0-9?)*) and slash (/): e.g. "MyName" or "example.com/MyName"
```
which is caused by name validation. (more than one slash)
https://github.com/kubernetes/kubernetes/blob/master/pkg/api/validation/validation.go#L92
https://github.com/kubernetes/kubernetes/blob/master/pkg/util/validation/validation.go#L38
related https://github.com/kubernetes/kubernetes/pull/25324

### PR DESCRIPTION
**Pull request by sttts**
Fixes #26610.

/cc @mdshuai @ncdc @jfrazelle

### CODE REVIEW

**Comment by ncdc** [comment]
@pmorie

File: docs/design/seccomp.md [hunk 1]
```diff
@@ -202,11 +202,11 @@ use annotations instead of extending the API with new fields.
 In the alpha version of this feature we will use annotations to store the
 names of seccomp profiles.  The keys will be:
 
-`security.alpha.kubernetes.io/seccomp/container/<container name>`
+`container.seccomp.security.alpha.kubernetes.io/<container name>`
 
 which will be used to set the seccomp profile of a container, and:
 
-`security.alpha.kubernetes.io/seccomp/pod`
+`seccomp.security.alpha.kubernetes.io/pod`
```
Comment (ncdc) [inline docs/design/seccomp.md]: would it make more sense to have this be pod.seccomp.security.alpha.kubernetes.io to be consistent with the container format?

Comment (sttts) [inline docs/design/seccomp.md]: we need a name (after the /)

Comment (sttts) [inline docs/design/seccomp.md]: but I agree that it looks asymmetric now. Suggestions welcome.

Comment (sttts) [inline docs/design/seccomp.md]:
```
	//     label-key ::= prefixed-name | name
	//     prefixed-name ::= prefix '/' name
	//     prefix ::= DNS_SUBDOMAIN
	//     name ::= DNS_LABEL
```

Comment (pmorie) [inline docs/design/seccomp.md]: I think what @sttts has proposed is probably what we have to do if only a single slash is allowed.  I would rather not do something like `/container-foo` and parse it.

Comment (ncdc) [inline docs/design/seccomp.md]: Yeah, I wasn't suggesting /container-foo. I was hoping to be consistent by having both pod and container be part of the prefix, but validation won't allow for that.

**Comment by ncdc** [comment]
Setting release-note-none as this PR just fixes a bug in seccomp support, which is new for 1.3.

**Comment by ncdc** [comment]
@k8s-bot unit test this issue: #26712

**Comment by ncdc** [comment]
cc @erictune @kubernetes/rh-cluster-infra

**Comment by pmorie** [comment]
@sttts were you going to add integration tests that establish that the labels are accepted?

**Comment by pmorie** [comment]
LGTM otherwise.

**Comment by sttts** [comment]
I have added e2e tests which check /proc/self/status. These will only work with Docker 1.10+ though.

File: test/e2e/security_context.go [hunk 2]
```diff
@@ -106,6 +107,37 @@ var _ = framework.KubeDescribe("Security Context [Feature:SecurityContext]", fun
 		testPodSELinuxLabeling(f, false, true)
 	})
 
+	It("should support seccomp alpha unconfined annotation on the container", func() {
```
Comment (pmorie) [inline test/e2e/security_context.go]: I think these should have `[Feature:Seccomp]` in the name

**Comment by pmorie** [comment]
@sttts thanks a lot for doing this; one nit and LGTM.  Want to make the PodSecurityPolicy changes in a follow-up?

**Comment by sttts** [comment]
Added `[Feature:Seccomp]`.

**Comment by sttts** [comment]
@k8s-bot test this issue: #23545

**Comment by sttts** [comment]
@k8s-bot test this issue: #23545

**Comment by sttts** [comment]
@k8s-bot test this issue: #23545

**Comment by ncdc** [comment]
@k8s-bot node e2e test this issue: #26431

**Comment by sttts** [comment]
@k8s-bot test this issue: #26431

**Pull request merged**